feeed/entropies.py

```diff
@@ -13,76 +17,225 @@ def __init__(self, feature_names='entropies'):
-    def entropy_lempel_ziv(cls, log_path):
-        return Entropies.default_call(log_path, "-z", "", "")
+    def entropy_lempel_ziv(cls, log):
+        unique_traces = [tuple(event["concept:name"] for event in trace) for trace in log]
+        N, N_w, words = 0, 0, set()
+
+        for trace in unique_traces:
+            word = ""
+            for activity in trace:
+                word += activity
+                if word not in words:
+                    words.add(word)
+                    word = ""
+
+            N += len(trace)
+
+        N_w = len(words)
+        
+        lempel_zev_entropy = N_w * math.log2(N) / N
+        return round(lempel_zev_entropy,3)
+    
+    @classmethod
+    def entropy_lempel_ziv_flattened(cls,log):
+        unique_traces = list(variants_filter.get_variants(log))
+        N, N_w, words = 0, 0, set()
+
+        for trace in unique_traces:
+            word = ""
+            for activity in trace:
+                word += activity
+                if word not in words:
+                    words.add(word)
+                    word = ""
+
+            N += len(trace)
+
+        N_w = len(words)
+        
+        lempel_zev_entropy = N_w * math.log2(N) / N
+        return round(lempel_zev_entropy,3)
+        
 
     @classmethod
-    def entropy_k_block_diff_1(cls, log_path):
-        return Entropies.default_call(log_path, "-d", "1", "")
+    def entropy_k_block_diff_1(cls, log,k=1):
+        return round(Entropies.entropy_k_block(log, k) - Entropies.entropy_k_block(log, k-1),3)
 
     @classmethod
-    def entropy_k_block_diff_3(cls, log_path):
-        return Entropies.default_call(log_path, "-d", "3", "")
+    def entropy_k_block_diff_3(cls, log,k=3):
+        return round(Entropies.entropy_k_block(log, k) - Entropies.entropy_k_block(log, k-1),3)
 
     @classmethod
-    def entropy_k_block_diff_5(cls, log_path):
-        return Entropies.default_call(log_path, "-d", "5", "")
+    def entropy_k_block_diff_5(cls, log,k=5):
+        return round(Entropies.entropy_k_block(log, k) - Entropies.entropy_k_block(log, k-1),3)
 
     @classmethod
-    def entropy_k_block_ratio_1(cls, log_path):
-        return Entropies.default_call(log_path, "-r", "1", "")
+    def entropy_k_block_ratio_1(cls, log,k=1):
+        return round(Entropies.entropy_k_block(log, k)/k,3)
 
     @classmethod
-    def entropy_k_block_ratio_3(cls, log_path):
-        return Entropies.default_call(log_path, "-r", "3", "")
+    def entropy_k_block_ratio_3(cls, log,k=3):
+        return round(Entropies.entropy_k_block(log, k)/k,3)
 
     @classmethod
-    def entropy_k_block_ratio_5(cls, log_path):
-        return Entropies.default_call(log_path, "-r", "5", "")
+    def entropy_k_block_ratio_5(cls, log,k=5):
+        return round(Entropies.entropy_k_block(log, k)/k,3)
 
     @classmethod
-    def entropy_knn_3(cls, log_path):
-        return Entropies.default_call(log_path, "-k", "3", "1")
+    def entropy_knn_3_flattened(cls, log,k=3):
```

Comment:
Please keep original name

Reply:
The values that java computes (along with the python implementation) is the flattened version. Would it still be appropriate to name it non-flattened?